test/cg.jl

```diff
@@ -50,6 +50,37 @@ Random.seed!(1234321)
         x0 = cg(A, zeros(T, n))
         @test x0 == zeros(T, n)
     end
+
+    @testset "Matrix{$T}, unconjugated dot product" for T in (Float32, Float64, ComplexF32, ComplexF64)
+        A = rand(T, n, n)
+        A = A + transpose(A) + 15I
+        x = ones(T, n)
+        b = A * x
+
+        reltol = √eps(real(T))
+
+        # Solve without preconditioner
+        x1, his1 = cocg(A, b, reltol = reltol, maxiter = 100, log = true)
+        @test isa(his1, ConvergenceHistory)
+        @test norm(A * x1 - b) / norm(b) ≤ reltol
+
+        # With an initial guess
+        x_guess = rand(T, n)
+        x2, his2 = cocg!(x_guess, A, b, reltol = reltol, maxiter = 100, log = true)
+        @test isa(his2, ConvergenceHistory)
+        @test x2 == x_guess
+        @test norm(A * x2 - b) / norm(b) ≤ reltol
+
+        # The following tests fails CI on Windows and Ubuntu due to a
+        # `SingularException(4)`
```

Comment:
What's going on with this failure?

Reply:
Actually the enclosing `@testset` of this `@test` is copied from [L13-L43](https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/blob/master/test/bicgstabl.jl#L13-L43) of `test/bicgstabl.jl`.  I didn't pay too much attention to the copied tests.  Let me try to run the tests on a Windows box and get back to you.

Reply:
I did `pkg> test IterativeSolvers` in a Windows box with the `continue` block ([L35-L37](https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/blob/master/test/bicgstabl.jl#L35-L37) of `test/bicgstabl.jl`) commented out, and all the tests run fine.  I will push the commit without the `continue` block in `test/cg.jl`.  Let's see if the tests succeed...

Reply:
No failure during CI (see below).  I will probably open another PR to remove the `continue` block from `test/bicgstabl.jl`.